test/parse.js

```diff
@@ -215,6 +215,16 @@ test('query strings having a brackets+separator array and format option as `brac
 	}), {foo: ['']});
 });
 
+test('query strings having a brackets+separator array and format option as `bracket-separator` with a URL encoded value', t => {
+	t.deepEqual(queryString.parse('?testA%5B%5D=1&testB%5B%5D=a%2Cb%2Cc%2Cd%2Ce%2Cf&testC=true', {
```

Comment:
I would prefer the string to be unencoded, and encode it before passing it to `parse`. This makes it possible to actually see what it contains when reading the code.